ctapipe/reco/HillasReconstructor.py

```diff
@@ -108,9 +108,11 @@ class that reconstructs the direction of an atmospheric shower
     def __init__(self, config=None, parent=None, **kwargs):
         super().__init__(config=config, parent=parent, **kwargs)
         self.hillas_planes = {}
+        self.divergent_mode = False
+        self.corrected_angle_dict = {}
 
-    def predict(self, hillas_dict, inst, pointing_alt, pointing_az):
-        '''
+    def predict(self, hillas_dict, inst,  array_pointing, telescopes_pointings=None):
```

Comment:
I'd suggest updating the base class as well either to have the same signature, or to remove this as an abstract function. In the former case, it might also imply a change to `ImPACTReconstructor` to have it use the same basic arguments. In that case it's not `hillas_dict` but `image_dict` or something, so maybe coming up with a more generic name would allow all to use the same interface (`telescope_features_dict` or something)

Reply:
Discussion started in #924 . Shall we continue there?
I might only say that by now ImPACT is a bit different so we might think of its refactoring later on.

Reply:
@kosack what do you think about this?

Reply:
yes, we can deal with that later